test/SlowTests/Server/Rachis/Cluster.cs

```diff
@@ -18,35 +19,31 @@ public class Cluster: ClusterTestBase
         public async Task test()
         {
             NoTimeouts();
-            //var leader = await CreateRaftClusterAndGetLeader(3);
-            var leader = await CreateRaftClusterAndGetLeader(2);
+            var leader = await CreateRaftClusterAndGetLeader(3);            
             CreateDatabaseResult databaseResult;
+            var replicationFactor = 2;
             using (var store = new DocumentStore()
             {
                 Url = leader.WebUrls[0],
                 DefaultDatabase = "test"
             }.Initialize())
             {
                 var doc = MultiDatabase.CreateDatabaseDocument("test");
-                databaseResult = store.Admin.Server.Send(new CreateDatabaseOperation(doc, replicationFactor: 2));
+                databaseResult = store.Admin.Server.Send(new CreateDatabaseOperation(doc, replicationFactor));
             }
-    
-            foreach (var server in Servers.Except(new [] {leader}))
-            {
-                await server.ServerStore.Cluster.WaitForIndexNotification(databaseResult.ETag ?? 0);
-
-                using (var store = new DocumentStore() { Url = server.WebUrls[0] }.Initialize())
-                using(var session = store.OpenAsyncSession())
+            int numberOfInstances = 0;
+            foreach (var server in Servers.Where(s => databaseResult.Topology.RelevantFor(s.ServerStore.NodeTag)))
+            {                
+                await server.ServerStore.WaitForCommitIndexChange(RachisConsensus.CommitIndexModification.Equal, databaseResult.ETag ?? 0);
```

Comment:
You are probably right, the thing is, we need to either change the implementation of the method or change its name because it isn't doing what the name implies